0-bootstrap/sa.tf

```diff
@@ -30,6 +30,11 @@ locals {
     "roles/browser", // Required for gcloud beta terraform vet to be able to read the ancestry of folders
   ]
 
+  on_prem_connection_roles = var.enable_on_prem_connection_roles ? [
+    "roles/secretmanager.viewer", // Required roles for retrieving secrets for VPN connection to on-prem if VPN enabled in network step
+    "roles/secretmanager.secretAccessor",
```

Comment:
Lets accept secret name instead of bool flag `enable_on_prem_connection_roles` and use granular IAM here since this will grant access to all secrets otherwise.

Reply:
fixed